example_app/src/app.rs

```diff
@@ -8,37 +11,44 @@ pub struct ExampleApp {
     frontend: Option<FrontEnd>,
 }
 
-impl epi::App for ExampleApp {
-    fn name(&self) -> &str {
-        "ewebsocket example app"
+impl ExampleApp {
+    pub fn new(cc: &eframe::CreationContext<'_>) -> Self {
+        let mut example_app = ExampleApp {
+            url: "wss://echo.websocket.events/.ws".into(),
+            ..Default::default()
+        };
+
+        example_app.connect(cc.egui_ctx.clone());
+
+        Self::default()
     }
 
-    fn setup(
-        &mut self,
-        _ctx: &egui::Context,
-        frame: &epi::Frame,
-        _storage: Option<&dyn epi::Storage>,
-    ) {
-        if let Some(web_info) = &frame.info().web_info {
-            // allow `?url=` query param
-            if let Some(url) = web_info.location.query_map.get("url") {
-                self.url = url.clone();
```

Comment:
This previous logic looked into `frame` to figure out whether we are in a web environment or not. However, with the use of the `new()` method to instantiate this app, it doesn't seem to have access to the frame information. Because of this, I'm not sure how to go about getting the information from before.

I know `update` gives us access to the frame, but I don't imagine putting this logic in there to be run every frame would be the best.

How might this logic be migrated the best?